test/e2e/framework/test_context.go

```diff
@@ -436,4 +439,70 @@ func AfterReadingAllFlags(t *TestContextType) {
 		}
 		os.Exit(1)
 	}
+
+	// Check for num-nodes if not set by looking for all nodes that are schedulable
+	if t.CloudConfig.NumNodes < 0 {
+		numNodes, err := countScheduleableNodes(t.KubeContext, t.KubeConfig, t.Host)
+		if err != nil {
+			klog.Errorf("Error getting num-nodes from API: %v", err)
+		} else {
+			t.CloudConfig.NumNodes = numNodes
```

Comment:
optionally omit the `} else {` above as `countScheduleableNodes` will return -1.
also i think we should check like so:
```
if t.CloudConfig.NumNodes == -1 {
```
instead of handling all negative integers.

also we must add the `-1` case as part of the `--num-nodes` flag description.
e.g.
> If the default value of "-1" is used the number of schedulable nodes is auto-detected."

Reply:
:+1: